utilities/fault_injection_env.cc

```diff
@@ -107,6 +107,51 @@ Status TestDirectory::Fsync() {
   return dir_->Fsync();
 }
 
+TestRandomAccessFile::TestRandomAccessFile(
+    std::unique_ptr<RandomAccessFile>&& target, FaultInjectionTestEnv* env)
+    : target_(std::move(target)), env_(env) {
+  assert(target_);
+  assert(env_);
+}
+
+Status TestRandomAccessFile::Read(uint64_t offset, size_t n, Slice* result,
+                                  char* scratch) const {
+  assert(env_);
+  if (!env_->IsFilesystemActive()) {
+    return env_->GetError();
+  }
+
+  assert(target_);
+  return target_->Read(offset, n, result, scratch);
+}
+
+Status TestRandomAccessFile::Prefetch(uint64_t offset, size_t n) {
+  assert(env_);
+  if (!env_->IsFilesystemActive()) {
+    return env_->GetError();
+  }
+
+  assert(target_);
+  return target_->Prefetch(offset, n);
+}
+
+Status TestRandomAccessFile::MultiRead(ReadRequest* reqs, size_t num_reqs) {
+  assert(env_);
+  if (!env_->IsFilesystemActive()) {
+    const Status s = env_->GetError();
+
+    assert(reqs);
+    for (size_t i = 0; i < num_reqs; ++i) {
+      reqs[i].status = s;
```

Comment:
Maybe better assert the size of `reqs` is always larger than `i`

Reply:
It's not really possible to do that with this interface: given only a pointer, there's no way of getting the "real" size of `reqs`. We have to rely on what the caller passes in `num_reqs`.

Reply:
> It's not really possible to do that with this interface: given only a pointer, there's no way of getting the "real" size of `reqs`. We have to rely on what the caller passes in `num_reqs`.

Oh, you are right.